toolchain/gcc.py

```diff
@@ -0,0 +1,120 @@
+            packages.extend(['glibc-devel-static',
+                             'zlib-devel',
+                             'elfutils',
+                             'libelf-devel',
+                             'gcc-c++',
+                             'isl-devel',
+                             'gmp-devel',
+                             'glibc-devel',
+                             'mpfr-devel',
+                             'makeinfo',
+                             'texinfo',
+                             'mpc-devel'])
+        elif dist.name == 'redhat':
+            packages.extend(['glibc-static',
+                             'autogen',
+                             'guile',
+                             'guile-devel',
+                             'libgo',
+                             'libgo-devel',
+                             'libgo-static',
+                             'elfutils-devel',
+                             'texinfo-tex',
+                             'texinfo',
+                             'elfutils-libelf-devel',
+                             'gmp-devel',
+                             'mpfr-devel',
+                             'libmpc-devel',
+                             'gcc-gnat',
+                             'libgnat'])
+        else:
+            self.log.warn("Unsupported OS!")
+
+        for package in packages:
+            if not sm.check_installed(package) and not sm.install(package):
+                self.error(
+                    "Failed to install %s required for this test." % package)
+        if dist.name == 'SuSE':
+            gcc_version = process.system_output("gcc --version")
+            get_system_gcc_version = re.search(
+                'gcc \(SUSE Linux\) (.*)\n',
+                gcc_version).group(1).replace(
+                ".",
+                "_")
+        else:
+            get_system_gcc_version = process.system_output(
+                "gcc -dumpversion").replace(".", "_").split("\n", 1)[0]
+        tarball = self.fetch_asset(
+            "https://github.com/gcc-mirror/gcc/archive/gcc-%s-release.zip" %
+            get_system_gcc_version)
+        archive.extract(tarball, self.srcdir)
+        gcc_folder = 'gcc-' + os.path.basename(tarball.split('.zip')[0])
+        self.srcdir = os.path.join(self.srcdir, gcc_folder)
+        os.chdir(self.srcdir)
+        process.run('./configure', ignore_status=True, sudo=True)
+        build.make(self.srcdir)
+
+    def test(self):
+        """
+        Runs the gcc `make check`
+        """
+        ret = process.run("make check", ignore_status=True, sudo=True)
+        logfile_name = os.path.join(self.logdir, "stdout")
+        logfile = open(logfile_name)
+        for line in logfile:
+            if line.startswith("\t\t==="):
+                self.log.info(line)
+            if line.startswith("#"):
+                self.log.info(line)
+        if ret.exit_status:
+            self.fail("Few gcc tests failed,refer the log file")
```

Comment:
can we parse log file as well and show fail point .Not any strong opinion but if we have this it will helpful